x-pack/plugins/cases/server/client/cases/get.ts

```diff
@@ -230,6 +232,86 @@ export const get = async (
   }
 };
 
+/**
+ * Retrieves a case resolving its ID and optionally loading its comments and sub case comments.
+ *
+ * @experimental
+ */
+export const resolve = async (
+  { id, includeComments, includeSubCaseComments }: GetParams,
+  clientArgs: CasesClientArgs
+): Promise<CaseResolveResponse> => {
+  const { unsecuredSavedObjectsClient, caseService, logger, authorization } = clientArgs;
+
+  try {
+    if (!ENABLE_CASE_CONNECTOR && includeSubCaseComments) {
+      throw Boom.badRequest(
+        'The `includeSubCaseComments` is not supported when the case connector feature is disabled'
+      );
+    }
+
+    const {
+      saved_object: savedObject,
+      ...resolveData
+    }: SavedObjectsResolveResponse<CaseAttributes> = await caseService.getResolveCase({
+      unsecuredSavedObjectsClient,
+      id,
+    });
+
+    await authorization.ensureAuthorized({
+      operation: Operations.resolveCase,
+      entities: [
+        {
+          id: savedObject.id,
+          owner: savedObject.attributes.owner,
+        },
+      ],
+    });
+
+    let subCaseIds: string[] = [];
+    if (ENABLE_CASE_CONNECTOR) {
+      const subCasesForCaseId = await caseService.findSubCasesByCaseId({
+        unsecuredSavedObjectsClient,
+        ids: [id],
+      });
+      subCaseIds = subCasesForCaseId.saved_objects.map((so) => so.id);
+    }
+
+    if (!includeComments) {
+      return CaseResolveResponseRt.encode({
+        ...resolveData,
+        case: flattenCaseSavedObject({
+          savedObject,
+          subCaseIds,
+        }),
+      });
+    }
+
+    const theComments = await caseService.getAllCaseComments({
+      unsecuredSavedObjectsClient,
+      id,
+      options: {
+        sortField: 'created_at',
+        sortOrder: 'asc',
+      },
+      includeSubCaseComments: ENABLE_CASE_CONNECTOR && includeSubCaseComments,
+    });
+
+    return CaseResolveResponseRt.encode({
+      ...resolveData,
+      case: flattenCaseSavedObject({
+        savedObject,
+        subCaseIds,
+        comments: theComments.saved_objects,
+        totalComment: theComments.total,
+        totalAlerts: countAlertsForID({ comments: theComments, id }),
+      }),
+    });
+  } catch (error) {
+    throw createCaseError({ message: `Failed to resolve case id: ${id}: ${error}`, error, logger });
```

Comment:
@cnasikas this will add an error in Kibana's log stating the authorization failure for a `getCase` operation.